src/upchunk.ts

```diff
@@ -1,9 +1,169 @@
 import { EventTarget, Event } from 'event-target-shim';
-import xhr, { XhrUrlConfig, XhrHeaders, XhrResponse } from 'xhr';
+import xhr, { type XhrUrlConfig, type XhrHeaders, type XhrResponse } from 'xhr';
+
+const DEFAULT_CHUNK_SIZE = 30720;
+const DEFAULT_MAX_CHUNK_SIZE = 512000; // in kB
+const DEFAULT_MIN_CHUNK_SIZE = 256; // in kB
+
+// Predicate function that returns true if a given `chunkSize` is valid, otherwise false.
```

Comment:
nit: this comment might be more helpful if it described what/why a valid chunk size is.

Reply:
FWIW what's valid here is whatever GCS allows 😆 From [the resumable upload docs](https://cloud.google.com/storage/docs/performing-resumable-uploads):

> The chunk size should be a multiple of 256 KiB (256 x 1024 bytes), unless it's the last chunk that completes the upload. Larger chunk sizes typically make uploads faster, but note that there's a tradeoff between speed and memory usage. It's recommended that you use at least 8 MiB for the chunk size.